src/components/Button.tsx

```diff
@@ -31,14 +31,67 @@ const Button = (
   const theme = useTheme();
   const buttonColor = color === 'danger' ? 'error' : color;
 
+  const iconPrimary = {
+    '&::before': {
+      backgroundColor: 'primary.dark',
+    },
+    '&:hover': {
+      backgroundColor: 'primaryStates.hover',
+    },
+    '&:active': {
+      backgroundColor: 'primaryStates.activeContained',
+    },
+    '&.Mui-disabled': {
+      backgroundColor: 'primaryStates.disabledBg',
+      color: 'primary.contrastText',
+    },
+  };
+
+  const iconSecondary = {
+    '&::before': {
+      backgroundColor: 'secondary.dark',
+    },
+    '&:hover': {
+      backgroundColor: 'secondaryStates.hover',
+    },
+    '&:active': {
+      backgroundColor: 'secondaryStates.activeContained',
+    },
+    '&.Mui-disabled': {
+      backgroundColor: 'secondaryStates.disabledBg',
+      color: 'secondary.contrastText',
+    },
+  };
+
+  const iconDanger = {
+    '&::before': {
+      backgroundColor: 'error.dark',
+    },
+    '&:hover': {
+      backgroundColor: 'errorStates.hover',
+    },
+    '&:active': {
+      backgroundColor: 'errorStates.activeContained',
+    },
+    '&.Mui-disabled': {
+      backgroundColor: 'errorStates.disabledBg',
+      color: 'error.contrastText',
+    },
+  };
+
   if (variant === 'icon') {
     return (
       <IconButton
         ref={ref}
         color={buttonColor}
         disabled={disabled}
         size="large"
-        sx={{ ...sx }}
+        sx={{
+          ...sx,
+          '&.MuiIconButton-colorPrimary': iconPrimary,
+          '&.MuiIconButton-colorSecondary': iconSecondary,
+          '&.MuiIconButton-colorError': iconDanger,
+        }}
```

Comment:
I  think that variable names would be cleaner with `Styles` suffix.
I'm not sure if that will be needed, but I would also spread `sx` after passing default styles, just in case someone needs to overwrite something.

```suggestion
        sx={{
          '&.MuiIconButton-colorPrimary': iconPrimaryStyles,
          '&.MuiIconButton-colorSecondary': iconSecondaryStyles,
          '&.MuiIconButton-colorError': iconDangerStyles,
           ...sx,
        }}
```

Reply:
Yeah, good catch with the spreading of the sx prop after. 👍